client/components/review-field.js

```diff
@@ -126,11 +128,34 @@ class ReviewField extends React.Component {
           : value
       }
 
+      const getChildren = value => {
+        const option = (options || []).find(option => option.value === value);
+        if (!option.reveal) {
+          return null;
+        }
+
+        return (
+          <div className="review-children">
+            <ReviewFields
+              fields={castArray(option.reveal).map(field => ({ ...field, preserveHeirarchy: true }))}
+              values={this.props.values}
+            />
+          </div>
+        )
+      }
+
       return (
         <ul>
           {
             value.filter(v => options ? options.find(o => o.value === v) : true).map(value => (
-              <li key={value}>{getValue(value)}</li>
+              <li key={value}>
+                {
+                  getValue(value)
+                }
+                {
+                  this.props.preserveHeirarchy && getChildren(value)
```

Comment:
What happens to these children if `preserveHierarchy` is false?

Reply:
Looks like it get's caught by `flattenReveals`

Reply:
yeah that, default is to flatten them to top level fields